### ISSUE TITLE
Hangs in fuzzy finder when there are many many files in the index

### ISSUE THREAD

**Issue by driskell** [file-icons/atom#592]
When opening fuzzy finder it hangs for over a second, and using arrow keys to move up and down also causes it to hang for over a second between each key press.

Profile below:

<img width="918" alt="screen shot 2017-04-25 at 20 35 14" src="https://cloud.githubusercontent.com/assets/939815/25425103/b027cdda-2a62-11e7-9a25-e0d337b41d4e.png">

The key part is the left side and right side with the "timeout" - these are times I was hitting the arrow keys. The middle bit is fuzzy finder performing a search.

Disabling file-icons removes the left/right long timeouts and makes the whole thing a bit more responsive and usable (especially when using arrow keys to scroll which becomes instant instead of seconds of delays.)

Appears the issue is down to here:
https://github.com/file-icons/atom/blob/master/lib/consumers/fuzzy-finder.js#L115
After every key press this triggers and the list contains approximately 80000 items and it seems to normalise the path of every item every time, even though only 10 of the items will become visible!

**Comment by Alhadis** [comment]
Well now, that's an interesting bottleneck.

@50Wliu So, any update on the five pull requests which need merging?

**Comment by Alhadis** [comment]
@driskell Sorry, I meant to give you a more detailed explanation on what's happening here.

The bottlenecking is an unfortunate side-effect of monkey-patching that was required to update icons dynamically. Such hackery was necessitated by current limitations of Atom's APIs: I'm still waiting on a [pull request](https://github.com/atom/fuzzy-finder/pull/281) to be reviewed that integrates the package *correctly*, which I predict will annihilate all current and future performance issues.

If the overhead is this palpable, you may be better off cloning [my fork](https://github.com/Cutlery-Drawer/fuzzy-finder) and linking it locally. Of course, that'd require Atom to be running in dev-mode...

**Bottom line:** This is hopefully temporary. It's ultimately up to the Atom team to decide what they want to do with these PRs.

**Comment by hultberg** [comment]
@Alhadis I have been using your fork in 1.19 beta on both macOS and linux, and I have not observed any issues. :+1:

**Comment by dvsekhvalnov** [comment]
Hey all, is there are any update on this issue? Experiencing it as well on atom 1.22.1  . Seems aforementioned pull request was merged.

**Comment by Alhadis** [comment]
@dvsekhvalnov It has been merged. We're just waiting for the next version of Atom to be released. It should be pretty soon I believe.

**Comment by dvsekhvalnov** [comment]
Nice, thanks for update !

**Comment by dvsekhvalnov** [comment]
Hey, just wanted to give feedback back on it. Updated to newest Atom v1.23.0 and fuzzy-finder not lagging any more on search. 

Awesome project guys, love it :)

**Comment by Alhadis** [comment]
@driskell Has the slowness disappeared on your end after updating?

**Comment by Alhadis** [comment]
Going from collective user feedback, I'm marking this issue as resolved, as no reports of lagging have been mentioned since Atom v1.23.0 hit stable. @driskell, please let me know if you're still experiencing this issue after upgrading, and I'll reopen this to investigate.
